### PR TITLE
VEC-413 build asvec in a docker container

### PR DESCRIPTION
**Pull request by arrowplum**
Adds Docker for multiarch asvec project along with GHA to build and deploy docker to jfrog with build info.

For simplicity made the Dockerfile at the project base. The version file to set the default base version.

### CODE REVIEW

File: .github/workflows/build-docker.yml [hunk 1]
```diff
@@ -0,0 +1,143 @@
+name: Build and Push Docker Image
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - 'v*.*.*'
```
Comment (dwelch-spike) [inline .github/workflows/build-docker.yml:8]:
```suggestion
      - '*.*.*'
```
We typically don't tag starting with `v`.

File: .github/workflows/build-docker.yml [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+      - name: Install JFrog CLI
+        uses: jfrog/setup-jfrog-cli@v3
+        with:
+          version: 'latest'
+
+      - name: Login to Artifact Aerospike Docker Registry
+        run: |
+            jf docker login artifact.aerospike.io --username ${{ secrets.JFROG_USERNAME }} --password ${{ secrets.JFROG_ACCESS_TOKEN }}
+            docker pull artifact.aerospike.io/ecosystem-container-dev-local/asvec:latest || true
+            docker push artifact.aerospike.io/ecosystem-container-dev-local/asvec:latest || true
```
Comment (dwelch-spike) [inline .github/workflows/build-docker.yml:47]:
```suggestion
            # needed these to get the docker cache working
            docker pull artifact.aerospike.io/ecosystem-container-dev-local/asvec:latest || true
            docker push artifact.aerospike.io/ecosystem-container-dev-local/asvec:latest || true
```
I think a comment about why this is needed would be good.

File: RELEASE [hunk 3]
```diff
@@ -0,0 +1 @@
+0.1.0-pre
```
Comment (dwelch-spike) [inline RELEASE:1]: In the future, we should either use this for all build actions, or the github tag.

**Review by dwelch-spike — COMMENTED** [review]
A few nit picks, after you have had a look this LGTM

File: .github/workflows/build-docker.yml (hunk 1 above)
Comment (arrowplum) [reply .github/workflows/build-docker.yml:8]: :thinking: we might maybe should but I will change it.

File: RELEASE (hunk 3 above)
Comment (arrowplum) [reply RELEASE:1]: Agreed.

**Review by dwelch-spike — APPROVED** [review]
LGTM

**Pull request merged**